File: src/main/java/com/deepak/grocery_delivery/service/AuthService.java
package com.deepak.grocery_delivery.service;

import com.deepak.grocery_delivery.dto.auth.LoginRequest;
import com.deepak.grocery_delivery.dto.auth.LoginResponse;
import com.deepak.grocery_delivery.dto.auth.RegisterRequest;
import com.deepak.grocery_delivery.dto.auth.RegisterResponse;

public interface AuthService {

    RegisterResponse register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

}

File: frontend/src/services/authService.js
import api from "./api";

export const loginUser = async (loginRequest) => {
    const response = await api.post(
        "/api/auth/login",
        loginRequest
    );

    return response.data;
};

export const registerUser = async (registerRequest) => {
    const response = await api.post(
        "/api/auth/register",
        registerRequest
    );

    return response.data;
};